test/python/qdmi/test_fomac.py

```diff
@@ -0,0 +1,369 @@
+# Copyright (c) 2023 - 2025 Chair for Design Automation, TUM
+# Copyright (c) 2025 Munich Quantum Software Company GmbH
+# All rights reserved.
+#
+# SPDX-License-Identifier: MIT
+#
+# Licensed under the MIT License
+
+"""Test the quantum computation IR."""
+
+from __future__ import annotations
+
+from typing import cast
+
+import pytest
+
+from mqt.core.qdmi.fomac import Device, DeviceStatus, Operation, Site, devices
+
+
+@pytest.fixture(params=devices())
+def device(request: pytest.FixtureRequest) -> Device:
+    """Fixture to provide a device for testing.
+
+    Returns:
+        Device: A quantum device instance.
+    """
+    return cast("Device", request.param)
+
+
+@pytest.fixture(params=devices())
+def device_and_site(request: pytest.FixtureRequest) -> tuple[Device, Site]:
+    """Fixture to provide a device for testing.
+
+    Returns:
+        tuple[Device, Site]: A tuple containing a quantum device instance and one of its sites.
+    """
+    device = request.param
+    site = device.sites()[0]
+    return device, site
+
+
+@pytest.fixture(params=devices())
+def device_and_operation(request: pytest.FixtureRequest) -> tuple[Device, Operation]:
+    """Fixture to provide a device for testing.
+
+    Returns:
+        tuple[Device, Operation]: A tuple containing a quantum device instance and one of its operations.
+    """
+    device = request.param
+    operation = device.operations()[0]
+    return device, operation
+
+
+def testdevice_name(device: Device) -> None:
```

Comment:
very minor nitpick: I would appreeciate a `_` between the word "test" and the actual test name here.